### PR TITLE
Implement Delete function, fix #58

### LINKED ISSUE #58: Delete and Expunge Feature
I would like to implement `delete(uids)` as a convenience function to `addFlags(uids, '\\Deleted')`
Moreover I would expose the `expunge()` function. 

Any tips what to look out for? Can I start?

I am using this in my new application (https://github.com/synox/void-mail). So far i just use connection.imap to access the underlying implementation, but this is not clean. 👍

### PR DESCRIPTION
**Pull request by synox**

### CODE REVIEW

File: lib/imapSimple.js [hunk 1]
```diff
@@ -384,6 +384,38 @@ ImapSimple.prototype.delFlags = function (uid, flags, callback) {
     });
 };
 
+/**
+ * Deletes the specified message(s).
+ *
+ * @param {string|Array} uid The messages uid
```
Comment (chadxz) [inline lib/imapSimple.js]: Try this for description: "The uid or array of uids indicating the messages to be deleted"

**Review by chadxz — CHANGES_REQUESTED** [review]

**Comment by chadxz** [comment]
Will you also add a section to the README for the method?

**Comment by synox** [comment]
yes ill do readme

**Comment by synox** [comment]
Do you also want to uupdate the example "Open messages and delete them"?
Done for today, it's late here.

**Comment by chadxz** [comment]
That'd be great

**Comment by synox** [comment]
I would also need a search function that does not fetch, often I  just need the uids. 

E.g. i want delete all mails older than 1 month, I don't want to fetch them first. Would you expose a search-without-fetch method or create a search+delete method?

So far i am doing for deleting:

			let searchCriteria = [
				['!DELETED'],
				['BEFORE', moment().subtract(30, 'days').toDate()]
			];
		const imapUnderlying = this.connection.imap
		const uids = await new Promise((resolve, reject) => {
			imapUnderlying.search(searchCriteria, (err, uids) => {
				if (err) {
					reject(err)
				} else {
					resolve(uids || [])
				}
			})
		})
		if (uids.length === 0) {
			return
		}
		await this.connection.deleteMessage(uids)

**Comment by chadxz** [comment]
Either one sounds reasonable. If you wanted a search without fetch we could make the search take an optional parameter to skip the fetch. Or we could add a deleteWhere that accepts search criteria.

**Comment by synox** [comment]
done, you can merge

**Comment by tennox** [comment]
@chadxz Would be great if you could merge this, so it could be released - and also #49

**Comment by chadxz** [comment]
Released as v5.0.0

Thanks for the nudge @TeNNoX

**Pull request merged**